src/NuGet.Core/NuGet.Protocol/HttpSource/HttpSource.cs

```diff
@@ -509,5 +511,22 @@ public void Dispose()
                 }
             }
         }
+
+        private void ThrowIfInsecureUri(string uri)
+        {
+            var parsedUri = new Uri(uri);
+
+            if (string.Equals(parsedUri.Scheme, Uri.UriSchemeHttp, StringComparison.OrdinalIgnoreCase))
+            {
+                if (_packageSource == null || !_packageSource.AllowInsecureConnections)
+                {
+                    throw new HttpSourceException(
+                        string.Format(
+                            CultureInfo.CurrentCulture,
+                            Strings.Error_Insecure_HTTP,
+                            _sourceUri.AbsoluteUri ?? "<unknown>", uri));
```

Comment:
Almost didn't notice there were 2 URIs here.

```suggestion
                            _sourceUri.AbsoluteUri ?? "<unknown>",
                            uri));
```